chapters/high_level_shader_language_comparison.adoc

```diff
@@ -56,15 +72,89 @@ HLSL:
 float4x3 mat = (float4x3)(ubo.view);
 ----
 
+Slang:
+[source,slang]
+----
+float4x3 mat = (float4x3)(ubo.view);
+// Or using generics
+let mat = cast<float4x3>(ubo.view);
+----
+
 [NOTE]
 ====
-An important difference: Matrices in GLSL are column-major, while matrices in HLSL are row-major. This affects things like matrix construction.
+An important difference: Matrices in GLSL are column-major, while matrices in HLSL and Slang are row-major by default. This affects things like matrix construction.
 ====
 
+=== Slang-specific Types
+
+Slang extends HLSL's type system with several advanced features:
+
+==== Interfaces
+
+Slang supports interfaces similar to C# or Java:
+
+[source,slang]
+----
+interface IMaterial {
+    float3 evaluateBRDF(float3 viewDir, float3 lightDir, float3 normal);
+}
+
+struct LambertianMaterial : IMaterial {
+    float3 albedo;
+
+    float3 evaluateBRDF(float3 viewDir, float3 lightDir, float3 normal) {
+        return albedo * max(0, dot(normal, lightDir)) / 3.14159;
+    }
+}
+----
+
+==== Generics
+
+Slang provides full generic support:
+
+[source,slang]
+----
+// Generic function
+generic<T>
+T min(T a, T b) {
```

Comment:
This and a couple other cases here appear to be using an old version of the generic syntax. I think this should work:
```slang
T min<T>(T a, T b) {
    return a < b ? a : b;
}
```